contracts/SafeDrop.vy

```diff
@@ -0,0 +1,38 @@
+# @version 0.3.10
+
+merkle_root: public(bytes32)
+
+struct Claim:
+  receiver: address
+  amount: uint256
+
+event ClaimMinted:
+  claim: Claim
+
+@external
+def __init__(merkle_root: bytes32):
+    self.merkle_root = merkle_root
+
+@external
+def mint(proof: DynArray[bytes32, 1024]):
+    claim: Claim = Claim({receiver: msg.sender, amount: 0})
+    assert self._verify(claim, proof)
```

Comment:
Don't you want to store this somewhere?

Reply:
yes I just haven't gotten to that part yet, still trying to successfully verify.